extensions/wikia/DefaultMessages/DefaultMessagesCache.php

```diff
@@ -0,0 +1,521 @@
+		}
+	}
+
+	private function memcKey( /*... */ ) {
+		global $wgDefaultMessagesDB;
+		$args = func_get_args();
+		$key = $wgDefaultMessagesDB . ':' . implode( ':', $args );
+		return $key;
+	}
+
+	/**
+	 * Try to load the cache from a local file.
+	 * Actual format of the file depends on the $wgLocalMessageCacheSerialized
+	 * setting.
+	 *
+	 * @param $hash String: the hash of contents, to check validity.
+	 * @param $code Mixed: Optional language code, see documenation of load().
+	 * @return false on failure.
+	 */
+	function loadFromLocal( $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$filename = $this->mLocalMessageCache . '/' . substr( $wgDefaultMessagesDB, 0, 1 ) . '/' . $wgDefaultMessagesDB . "/messages-$code";
+
+		# Check file existence
+		wfSuppressWarnings();
+		$file = fopen( $filename, 'r' );
+		wfRestoreWarnings();
+		if ( !$file ) {
+			return false; // No cache file
+		}
+
+		if ( $this->mLocalMessageCacheSerialized ) {
+			// Check to see if the file has the hash specified
+			$localHash = fread( $file, 32 );
+			if ( $hash === $localHash ) {
+				// All good, get the rest of it
+				$serialized = '';
+				while ( !feof( $file ) ) {
+					$serialized .= fread( $file, 100000 );
+				}
+				fclose( $file );
+				return $this->setCache( unserialize( $serialized ), $code );
+			} else {
+				fclose( $file );
+				return false; // Wrong hash
+			}
+		} else {
+			$localHash=substr(fread($file,40),8);
+			fclose($file);
+			if ($hash!=$localHash) {
+				return false; // Wrong hash
+			}
+
+			# Require overwrites the member variable or just shadows it?
+			require( $filename );
+			return $this->setCache( $this->mCache, $code );
+		}
+	}
+
+	/**
+	 * Save the cache to a local file.
+	 */
+	function saveToLocal( $serialized, $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$dirname = $this->mLocalMessageCache . '/' . substr( $wgDefaultMessagesDB, 0, 1 ) . '/' . $wgDefaultMessagesDB;
+		$filename = "$dirname/messages-$code";
+		wfMkdirParents( $dirname, 0777 ); // might fail
+
+		wfSuppressWarnings();
+		$file = fopen( $filename, 'w' );
+		wfRestoreWarnings();
+
+		if ( !$file ) {
+			wfDebug( "Unable to open local cache file for writing\n" );
+			return;
+		}
+
+		fwrite( $file, $hash . $serialized );
+		fclose( $file );
+		@chmod( $filename, 0666 );
+	}
+
+	function saveToScript( $array, $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$filename = $this->mLocalMessageCache . "/messages-$wgDefaultMessagesDB-$code";
+		$tempFilename = $filename . '.tmp';
+		wfMkdirParents( $this->mLocalMessageCache, 0777 ); // might fail
+
+		wfSuppressWarnings();
+		$file = fopen( $tempFilename, 'w');
+		wfRestoreWarnings();
+
+		if ( !$file ) {
+			wfDebug( "Unable to open local cache file for writing\n" );
+			return;
+		}
+
+		fwrite($file,"<?php\n//$hash\n\n \$this->mCache = array(");
+
+		foreach ($array as $key => $message) {
+			$key = $this->escapeForScript($key);
+			$messages = $this->escapeForScript($message);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$messages" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)